### PR TITLE
Bug fix - Use ASSET_PIPELINE in asset generator to fix Docker builds

### PR DESCRIPTION
**Pull request by ewlarson**
Restores Vite as the asset pipeline for Docker builds.

Fixes: #1512

### CODE REVIEW

File: template.rb [hunk 1]
```diff
@@ -14,5 +14,6 @@
 
 generate "blacklight:install", "--devise"
 generate "geoblacklight:install", "-f"
+generate "geoblacklight:assets:vite"
```
Comment (ewlarson) [inline template.rb:17]: Forcing the vite generator here does resolved the assets.

File: .github/workflows/build-demo-app.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ jobs:
         run: exec "yarnpkg"
 
       - name: Build
-        run: BRANCH=${{ env.BRANCH }} rails new build -m template.rb -a propshaft
+        run: BRANCH=${{ env.BRANCH }} ASSET_PIPELINE=vite rails new build -m template.rb -a propshaft
```
Comment (ewlarson) [inline .github/workflows/build-demo-app.yml:38]: Adding this env var here did propogate the value through the to docker build.

**Comment by ewlarson** [comment]
Assets via Vite:

Screenshot 2024-08-16 at 3 42 05 PM

**Review by thatbudakguy — CHANGES_REQUESTED** [review]
Vite is the default and shouldn't require any configuration:
https://github.com/geoblacklight/geoblacklight/blob/6776b5ac8c08c041733e76f7f3f4ce6a1d0dd71a/lib/generators/geoblacklight/assets_generator.rb#L7

Is something else messed up here? Is `ASSET_PIPELINE` somehow set to something in the docker container? You shouldn't need to set it at all in order for the vite generator to run. Maybe the `ENV.fetch` logic is somehow wrong?

**Comment by thatbudakguy** [comment]
I wonder if the `class_option` should just default to `vite`, and then we should use `ENV["ASSET_PIPELINE"]` explicitly in the template.rb file?

You should never need to call the asset generator directly; `geoblacklight:install` should invoke it for you.

**Comment by ewlarson** [comment]
> Is something else messed up here? Is ASSET_PIPELINE somehow set to something in the docker container? You shouldn't need to set it at all in order for the vite generator to run. Maybe the ENV.fetch logic is somehow wrong?

Yeah, your code _reads_ fine to me, but for whatever reason, the asset generator just doesn't run... This change is a bit heavy-handed, but fixes the current no-assets Docker builds.

**Review by eliotjordan — APPROVED** [review]

**Comment by eliotjordan** [comment]
@ewlarson @thatbudakguy There were a few problems with the install and assets generators - primarily the assets were not generated outside if :test. I fixed that up and was able to get it working without running the assets generator directly in template.rb.

**Review by eliotjordan — APPROVED** [review]

**Pull request merged**